packages/plugins/response-cache/src/plugin.ts

```diff
@@ -207,39 +208,9 @@ export type ResponseCacheExecutionResult = ExecutionResult<
 const originalDocumentMap = new WeakMap<DocumentNode, DocumentNode>();
 const addTypeNameToDocument = memoize2(function addTypeNameToDocument(
```

Comment:
I'm not sure why do we memoize this since we also check if the document has been parsed before calling it.

Reply:
What do you mean by checking if it has been parsed?

Reply:
I mean that we have this guard at the start of the `onSchemaChange` which prevent the execution of the function :
```ts
      if (processedSchemas.has(schema)) {
        visitor = processedSchemas.get(schema)!;
        return;
      }
```

And the only way this is true is if the function has already been called with this document, with this line at the end of `addTypeNameToDocument:

```ts
	originalDocumentMap.set(newDocument, document);
```

So I'm not sure we ever make a memoized call.

Reply:
If the document is cached by parser cache or sth else, this memoization allows us to avoid multiple visits over the document.
`originalDocumentMap` is a different map that keeps documents in other way around.

Reply:
Hummm I think I understand the memoization part, but not why we check for the presence in the map then. In which situation we could have the document being present as a key in the `originalDocumentMap` if the keys are the modified documents ?